genkit-tools/cli/context/GENKIT.md

```diff
@@ -0,0 +1,394 @@
+    const response = await ai.generate({
+      model: googleAI.model('imagen-3.0-generate-002'),
+      prompt,
+      output: { format: 'media' },
+    });
+
+    if (!response.media?.url) {
+      throw new Error('Image generation failed to produce media.');
+    }
+
+    const parsed = parseDataURL(response.media.url);
+    if (!parsed) {
+      throw new Error('Could not parse image data URL.');
+    }
+
+    const outputPath = './output.png';
+    await fs.writeFile(outputPath, parsed.body);
+    return outputPath;
+  }
+);
+```
+
+</example>
+
+<example>
+
+### Video Generation
+
+```ts
+import * as fs from 'fs';
+import { Readable } from 'stream';
+import { pipeline } from 'stream/promises';
+
+...
+
+export const videoGenerationFlow = ai.defineFlow(
+  {
+    name: 'videoGenerationFlow',
+    inputSchema: z
+      .string()
+      .describe('A detailed description for the video scene'),
+    outputSchema: z.string().describe('Path to the generated .mp4 video file'),
+  },
+  async (prompt) => {
+    let { operation } = await ai.generate({
+      model: googleAI.model('veo-3.0-generate-preview'),
+      prompt,
+    });
+
+    if (!operation) {
+      throw new Error('Expected the model to return an operation.');
+    }
+
+    console.log('Video generation started... Polling for completion.');
+    while (!operation.done) {
+      await new Promise((resolve) => setTimeout(resolve, 5000));
+      operation = await ai.checkOperation(operation);
+      console.log(
+        `Operation status: ${operation.done ? 'Done' : 'In Progress'}`
+      );
+    }
+
+    if (operation.error) {
+      throw new Error(`Video generation failed: ${operation.error.message}`);
+    }
+
+    const video = operation.output?.message?.content.find((p) => !!p.media);
+    if (!video?.media?.url) {
+      throw new Error(
+        'Failed to find the generated video in the operation output.'
+      );
+    }
+
+    const videoUrl = `${video.media.url}&key=${process.env.GEMINI_API_KEY}`;
+    const videoResponse = await fetch(videoUrl);
+
+    if (!videoResponse.ok || !videoResponse.body) {
+      throw new Error(`Failed to fetch video: ${videoResponse.statusText}`);
+    }
+
+    const outputPath = './output.mp4';
+    const fileStream = fs.createWriteStream(outputPath);
+    await pipeline(Readable.fromWeb(videoResponse.body as any), fileStream);
+
+    return outputPath;
+  }
+);
+```
+
+</example>
+
+---
+
+## Running and Inspecting Flows
+
+1.  **Add Build Script**: Add the following to `package.json`:
+
+    ```json
+    {
+      "scripts": {
+        "build": "tsc"
+      }
+    }
+    ```
+
+2.  **Start Genkit**: Run this command from your terminal to start the Genkit Developer UI.
+
+    ```bash
+    genkit start
+    ```
+
+    Then, in a separate terminal, run the build command in watch mode:
+
+    ```bash
+    npm run build -- --watch
+    ```
+
+    Visit [http://localhost:4000](http://localhost:4000) to inspect and run your flows.
+
```

Comment:
Same question: This can be project specific and we should probably omit this info, and just specify the `genkit start` command and the Dev UI.

Reply:
you can try removing it but i had various issues. everything in the file was added through trial and error. though i dont recall which things fixed what.